### DISCUSSION TITLE
Can set cpu limit and memory limit?

### DISCUSSION THREAD

**Discussion by zcl0621** [django/daphne#386] [Q&A]
I used daphne to run django channels in kubernetes pod. django running by uwsgi.

django == 2.2.7
daphne == 2.4.0
channels ==2.2.0
`
[uwsgi]
chdir = /project
module = backend.wsgi:application
master = True
processes = 1
http = 0.0.0.0:8000
vacuum = True
max-requests = 2000
enable-threads = True
thunder-lock = True
limit-as = 256
asyncio = 200
attach-daemon = celery -A backend beat -l info
attach-daemon = celery -A backend worker -l info --autoscale=15,10 -E
attach-daemon = daphne -b 127.0.0.1 -p 8001 backend.asgi:application
`
I set pod 1 core and 1G memory.

If the pod run on 8 core 8G host. The pod running good.
But if the pod run on 32 core 64G host. The daphne will memory error.

The daphne will for too many process and used too many memory.Then kill by uwsgi.

So done daphne can set cpu limit and memory limit or ready set by cgroup?

thanks!

**Comment by carltongibson** [comment]
This is something you'd do with Systemd or Docker or …

See https://www.redhat.com/sysadmin/cgroups-part-four for guide.
